modeldb/modeldb.py

```diff
@@ -57,6 +57,27 @@ def download_model(arg_tuple):
                 # if you need to test changes to a model that does not exist on
                 # GitHub under the ModelDBRepository organisation.
                 organisation = github[1:]
+            elif github.startswith("local:"):
+                # Using
+                #  github: "local: /path/to/repository"
+                # in modeldb-run.yaml implies that we 'git clone /path/to/repository'.
+                # This is useful to tentatively explore the effect of
+                # model changes on results with different nrn versions
+                # without committing to github or before being ready to
+                # make a pull request
+                print("\nTo be cloned by runmodel", github)
+                return model_id, model
+            elif github.startswith("copy:"):
+                # Using
+                #  github: "copy: /path/to/parentfolder"
+                # in modeldb-run.yaml implies that we
+                #  'cp -R /path/to/parentfolder/<id> <workingdir>'
+                # A copy differs from a clone in that local changes in
+                # the checkout are mirrored in the copy without having to
+                # be committed. Note the copy leaves out the .git and
+                # x86_64 folders.
+                print("\nTo be copied by runmodel %s/%s" % (github, model_id))
+                return model_id, model
```

Comment:
> I think the environment variable approach above export MODELDB_LOCAL_REPOSITORIES=$HOME/models/modeldb might be better than having to modify modeldb-meta.yaml. 

That's also OK. But this will also need local commits.

Anyway, I feel like the scope + impact of this decision is limited. So I think it's perfectly fine if you choose either approach.

Reply:
Important part is to have this flexibility and document it properly.

Reply:
I don't think there needs to be any hurry with this. I'll let it ripen for a while wait til the next round of modeldb updates to see how some of the details help with the workflow.